### ISSUE TITLE
verify sometimes fails

### ISSUE THREAD

**Issue by freqlabs** [ranisalt/node-argon2#107]
```javascript
argon2 = require('argon2');
argon2.hash("password")
    .then(hash => argon2.verify(hash, "password"))
    .then(console.log);
```
Running this in a loop I see `false` ~7% of the time.

### Environment
**Operating system**:
Tested on FreeBSD 11.1 and macOS 10.13.2

**Node version:**
v9.3.0 and v9.4.0

**Compiler version:**
FreeBSD clang version 4.0.0 (tags/RELEASE_400/final 297347) (based on LLVM 4.0.0)
Apple LLVM version 9.0.0 (clang-900.0.39.2)

**Comment by freqlabs** [comment]
I should add, this was tested with the latest tagged version, `0.17.0`.

**Comment by freqlabs** [comment]
Same problem on Linux 4.14.12-1-ARCH with node v9.4.0 and gcc 7.2.1

**Comment by ranisalt** [comment]
That's strange. I set the benchmark to test the result everytime and it does not fail a single time, running hundreds of hashings per second.

**Comment by eden** [comment]
FYI, I narrowed it down this commit: https://github.com/ranisalt/node-argon2/commit/5245bcf11bb55a3bd82c5a4a04a5f85f139a975f

It seems the OK callback has an issue with how it returned the result to the caller. I believe I have a fix in this pull request #114

**Comment by eden** [comment]
BTW, I hit this as well, and had to revert back to 0.16.2.
